### ISSUE TITLE
Error on init when using constant and literal

### ISSUE THREAD

**Issue by lbinria** [imitator-model-checker/imitator#50]
## Description

An error is raised when using a constant and a literal in the init section, because the literal is not inferred before the init expression trying to reduce.

Example : 
```
b := i = 0 with i = 0 : int;
```

## Solution

Just have to return inferred expression when doing the type checking on assignment. Then use this inferred expression when trying to reduce.

**Comment by lbinria** [comment]
Bug fixed
